qiskit/pulse/builder.py

```diff
@@ -0,0 +1,1986 @@
+        with pulse.align_sequential():
+            pulse.play(gaussian_pulse, d0)
+            # Play another pulse after at t=20.
+            pulse.play(gaussian_pulse, d1)
+
+            # We can also nest contexts as each instruction is
+            # contained in its local scheduling context.
+            # The output of a child context is a scheduled block
+            # with the internal instructions timing fixed relative to
+            # one another. This is block is then called in the parent context.
+
+            # Context starts at t=30.
+            with pulse.align_left():
+                # Start at t=30.
+                pulse.play(gaussian_pulse, d0)
+                # Start at t=30.
+                pulse.play(gaussian_pulse, d1)
+            # Context ends at t=40.
+
+            # Alignment context where all pulse instructions are
+            # aligned to the right, ie., as late as possible.
+            with pulse.align_right():
+                # Shift the phase of a pulse channel.
+                pulse.shift_phase(math.pi, d1)
+                # Starts at t=40.
+                pulse.delay(100, d0)
+                # Ends at t=140.
+
+                # Starts at t=130.
+                pulse.play(gaussian_pulse, d1)
+                # Ends at t=140.
+
+            # Acquire data for a qubit and store in a memory slot.
+            pulse.acquire(100, 0, pulse.MemorySlot(0))
+
+            # We also support a variety of macros for common operations.
+
+            # Measure all qubits.
+            pulse.measure_all()
+
+            # Delay on some qubits.
+            # This requires knowledge of which channels belong to which qubits.
+            # delay for 100 cycles on qubits 0 and 1.
+            pulse.delay_qubits(100, 0, 1)
+
+            # Call a quantum circuit. The pulse builder lazily constructs a quantum
+            # circuit which is then transpiled and scheduled before inserting into
+            # a pulse schedule.
+            # NOTE: Quantum register indices correspond to physical qubit indices.
+            qc = QuantumCircuit(2, 2)
+            qc.cx(0, 1)
+            pulse.call(qc)
+            # Calling a small set of standard gates and decomposing to pulses is
+            # also supported with more natural syntax.
+            pulse.u3(0, math.pi, 0, 0)
+            pulse.cx(0, 1)
+
+
+            # It is also be possible to call a preexisting schedule
+            tmp_sched = pulse.Schedule()
+            tmp_sched += pulse.Play(gaussian_pulse, d0)
+            pulse.call(tmp_sched)
+
+            # We also support:
+
+            # frequency instructions
+            pulse.set_frequency(5.0e9, d0)
+
+            # phase instructions
+            pulse.shift_phase(0.1, d0)
+
+            # offset contexts
+            with pulse.phase_offset(math.pi, d0):
+                pulse.play(gaussian_pulse, d0)
+
+The above is just a small taste of what is possible with the builder. See the
```

Comment:
Alternatively to breaking up the above, you can just remove it all. Everything below is more important.

Reply:
I would like to keep it but shrink and modify it into a more expository section as you described above. We should do this as part of the documentation PR pre-release.